### ISSUE TITLE
Rerun TCP protocol should be more structured and only log warnings if the incoming connection is from a Rerun SDK

### ISSUE THREAD

**Issue by jleibs** [rerun-io/rerun#5883]
Currently, the Rerun TCP protocol immediately starts with a version number.

Unfortunately, our port, 9876 is fairly low-entropy and likely to show up in the wild from non-Rerun sources or as part of security probes. This means we are quite likely to see incoming connections from non-Rerun sources.

Currently we Warn on protocol version mismatch:
```
[2024-04-09T11:23:43Z WARN  re_sdk_comms::server] Closing connection to client at 130.237.72.38:58008: SDK client is using a newer protocol version (790) than the SDK server (0)
```

However, sometimes these incoming connections are outside of our control.  The more principled approach would be to ALWAYS start with a rerun-unique header and then follow that with the version number. This way if the expected header isn't there we can ignore the connection and not produce the warning.

**Comment by gurry** [comment]
@emilk In case you haven't started, I have already created this fix.

**Comment by Wumpf** [comment]
I messed up unfortunately and removed this fix last-minute from 0.16 which as of the time of writing is in the process of being released by our CI job.
Documented here https://github.com/rerun-io/rerun/pull/6332/files#r1603677569
